cpp/src/arrow/ipc/reader.cc

```diff
@@ -926,6 +935,15 @@ Status MakeSparseTensorWithSparseCSRIndex(
   return Status::OK();
 }
 
+Status MakeSparseTensorWithSparseCSCIndex(
```

Comment:
Should this also change to `Result` type? I suppose that would require changing `ReadSparseTensorPayload` and many other functions, so maybe out of scope here.

Reply:
We should probably change all `MakeSparseTensorWithSparseXXXIndex` functions at the same time, for consistency. It can be done in another PR.

Reply:
I will change these functions to the `Result`-returning version in another pull-request.

Reply:
Filed https://issues.apache.org/jira/browse/ARROW-7420